doc: nrf-bm: lib: bluetooth: services: add ble HRS central (#583)
nrfconnect/sdk-nrf-bm: doc/nrf-bm/libraries/bluetooth/services/ble_hrs_central.rst

Heart Rate Service (HRS) Client
###############################

.. contents::
   :local:
   :depth: 2

This module implements the Heart Rate Service Client to retrieve Heart Rate Measurements and Body Sensor Location of a peripheral device with the :ref:`lib_ble_service_hrs`.

Overview
********

The HRS Client can retrieve information such as the sensor location from a device that provides the Heart Rate Service.

Configuration
*************

Set the :kconfig:option:`CONFIG_BLE_HRS_CENTRAL` Kconfig option to enable the service.

Set the maximum number of RR intervals per HRM notification with the :kconfig:option:`CONFIG_BLE_HRS_CENTRAL_RR_INTERVALS_MAX_COUNT` Kconfig option.

Initialization
==============

The service instance is declared using the :c:macro:`BLE_HRS_CENTRAL_DEF` macro, specifying the name of the instance.
The service is initialized by calling the :c:func:`ble_hrs_central_init` function.
See the :c:struct:`ble_hrs_central_config` struct for configuration details, in addition to the HRS specification.

Usage
*****

Applications can use the :ref:`lib_ble_scan` library for detecting advertising devices that support the Heart Rate Service.

> Upon connection, the application can use the :ref:`lib_ble_db_discovery` library to discover the peers database.

Reviewer: ```suggestion
Upon connection, the application can use the :ref:`lib_ble_db_discovery` library to discover the peer's database.
```

Call :c:func:`ble_hrs_on_db_disc_evt` on callback events from the Database Discovery.
This function sends the :c:enumerator:`BLE_HRS_CENTRAL_EVT_DISCOVERY_COMPLETE` event to the application if the Heart Rate Service is discovered in the peer's database.
The application must call the :c:func:`ble_hrs_central_handles_assign` function to associate the link with the HRS Client instance.

Upon Heart Rate Service discovery the application can call the :c:func:`ble_hrs_central_hrm_notif_enable` function to request the peer to start sending Heart Rate Measurement notifications.

Dependencies
************

This library uses the following |BMshort| libraries:

> * SoftDevice - :kconfig:option:`CONFIG_SOFTDEVICE`

Author: SoftDevice has it's separate documentation. We should probably  add something for the Kconfig though.

> * SoftDevice handler - :kconfig:option:`CONFIG_NRF_SDH`

Reviewer 2: ```suggestion
* :ref:`lib_nrf_sdh` - :kconfig:option:`CONFIG_NRF_SDH`
```

* Bluetooth LE DB Discovery - :kconfig:option:`CONFIG_BLE_DB_DISCOVERY`

The library is expected to be used with the following |BMshort| libraries:

> * Bluetooth LE Scan - :kconfig:option:`CONFIG_BLE_SCAN`

Reviewer 2: ```suggestion
The library is expected to be used with the :ref:`lib_ble_scan` |BMshort| library - :kconfig:option:`CONFIG_BLE_SCAN`
```

.. _lib_ble_service_hrs_central:

Heart Rate Service (HRS) Central
################################

.. contents::
   :local:
   :depth: 2

> This module implements the Heart Rate Service Central to retrieve information about the Heart Rate Measurement and Body Sensor Location of a peripheral device with the :ref:`lib_ble_service_hrs`.

Reviewer: ```suggestion
This module implements the Heart Rate Service Client to retrieve Heart Rate Measurements and Body Sensor Location of a peripheral device with the :ref:`lib_ble_service_hrs`.
```

Configuration
*************

Set the :kconfig:option:`CONFIG_BLE_HRS_CENTRAL` Kconfig option to enable the service.

Set the maximum number of RR intervals per HRM notification with the :kconfig:option:`CONFIG_BLE_HRS_CENTRAL_RR_INTERVALS_MAX_COUNT` Kconfig option.

Initialization
==============

The service instance is declared using the :c:macro:`BLE_HRS_CENTRAL_DEF` macro, specifying the name of the instance.
The service is initialized by calling the :c:func:`ble_hrs_central_init` function.
See the :c:struct:`ble_hrs_central_config` struct for configuration details, in addition to the HRS specification.

Usage
*****

Applications can use the :ref:`lib_ble_scan` library for detecting advertising devices that support the Heart Rate Service.

Upon connection, the application can use the :ref:`lib_ble_db_discovery` library to detect the peers database.
Call :c:func:`ble_hrs_on_db_disc_evt` on callback events from the Database Discovery.
This function determines whether the event relates to the discovery of Heart Rate Service at the peer.
If it does, the application's event handler is called to indicate that the Heart Rate Service was discovered at the peer.

> The application must call the :c:func:`ble_hrs_central_handles_assign` function to associate the link to the HRS Client instance.

Reviewer: ```suggestion
The application must call the :c:func:`ble_hrs_central_handles_assign` function to associate the link with the HRS Client instance.
```
